test/functional/plugin/lsp_spec.lua

```diff
@@ -6121,6 +6121,23 @@ describe('LSP', function()
     end)
   end)
 
+  describe('vim.lsp.start', function()
+    it('can launch when an existing client has no workspace_folders', function()
+      exec_lua(create_server_definition)
+      exec_lua(function()
+        local server = _G._create_server({
+          handlers = {
+            initialize = function(_, _, callback)
+              callback(nil, { capabilities = {} })
+            end,
+          },
+        })
+        vim.lsp.start { cmd = server.cmd, name = 'foo' }
+        vim.lsp.start { cmd = server.cmd, name = 'foo', root_dir = 'bar' }
+      end)
+    end)
```

Comment:
seems like this test is missing an assertion (such as `eq()`). does the test fail if you undo the change to `runtime/lua/vim/lsp.lua` ?

Reply:
> seems like this test is missing an assertion (such as eq()). does the test fail if you undo the change to runtime/lua/vim/lsp.lua ?

Yes, it failed.

Maybe it's always a better practice to assert more thing than minimal reproduce...